ops_test.go

```diff
@@ -0,0 +1,180 @@
+}
+
+func TestLast(t *testing.T) {
+	assert.Equal(t, Last([]int{1, 2, 3}), 3, "1, 2, 3")
+	assert.Equal(t, Last([]int{}), 0, "[]int{}")
+	assert.Equal(t, Last([]float64{1.0, 2.0, 3.0}), 3.0, "1.0, 2.0, 3.0")
+	assert.Equal(t, Last([]float64{}), 0.0, "[]float64{}")
+	assert.Equal(t, Last([]string{"a", "b", "c"}), "c", `"a", "b", "c"`)
+	assert.Equal(t, Last([]string{}), "", "[]string{}")
+}
+
+func TestZeroValue(t *testing.T) {
+	assert.Equal(t, ZeroValue[int](), 0, "int")
+	assert.Equal(t, ZeroValue[float64](), 0.0, "float64")
+	assert.Equal(t, ZeroValue[string](), "", "string")
+}
+
+func TestMap(t *testing.T) {
+	assert.Equal(t, Map(func(x int) int { return x * 2 }, []int{1, 2, 3}), []int{2, 4, 6}, "1, 2, 3")
+	assert.Equal(t, Map(func(x float64) float64 { return x * 2 }, []float64{1.0, 2.0, 3.0}), []float64{2.0, 4.0, 6.0}, "1.0, 2.0, 3.0")
+	assert.Equal(t, Map(func(x string) string { return x + x }, []string{"a", "b", "c"}), []string{"aa", "bb", "cc"}, `"a", "b", "c"`)
+	assert.Equal(t, Map(func(x int) int { return x * 2 }, []int{}), []int{}, "[]int{}")
+	assert.Equal(t, Map(func(x float64) float64 { return x * 2 }, []float64{}), []float64{}, "[]float64{}")
+	assert.Equal(t, Map(func(x string) string { return x + x }, []string{}), []string{}, "[]string{}")
+}
+
+func TestFilter(t *testing.T) {
+	assert.Equal(t, Filter(func(x int) bool { return x > 2 }, []int{1, 2, 3}), []int{3}, "1, 2, 3")
+	assert.Equal(t, Filter(func(x float64) bool { return x > 2.0 }, []float64{1.0, 2.0, 3.0}), []float64{3.0}, "1.0, 2.0, 3.0")
+	assert.Equal(t, Filter(func(x string) bool { return x > "b" }, []string{"a", "b", "c"}), []string{"c"}, `"a", "b", "c"`)
+	assert.Equal(t, Filter(func(x int) bool { return x > 2 }, []int{}), []int{}, "[]int{}")
+	assert.Equal(t, Filter(func(x float64) bool { return x > 2.0 }, []float64{}), []float64{}, "[]float64{}")
+	assert.Equal(t, Filter(func(x string) bool { return x > "b" }, []string{}), []string{}, "[]string{}")
+}
+
+func TestTake(t *testing.T) {
+	assert.Equal(t, Take(2, []int{1, 2, 3}), []int{1, 2}, "1, 2, 3")
+	assert.Equal(t, Take(2, []float64{1.0, 2.0, 3.0}), []float64{1.0, 2.0}, "1.0, 2.0, 3.0")
+	assert.Equal(t, Take(2, []string{"a", "b", "c"}), []string{"a", "b"}, `"a", "b", "c"`)
+	assert.Equal(t, Take(2, []int{}), []int{}, "[]int{}")
+	assert.Equal(t, Take(2, []float64{}), []float64{}, "[]float64{}")
+	assert.Equal(t, Take(2, []string{}), []string{}, "[]string{}")
+}
+
+func TestTail(t *testing.T) {
+	assert.Equal(t, Tail(2, []int{1, 2, 3}), []int{2, 3}, "1, 2, 3")
+	assert.Equal(t, Tail(2, []float64{1.0, 2.0, 3.0}), []float64{2.0, 3.0}, "1.0, 2.0, 3.0")
+	assert.Equal(t, Tail(2, []string{"a", "b", "c"}), []string{"b", "c"}, `"a", "b", "c"`)
+	assert.Equal(t, Tail(2, []int{}), []int{}, "[]int{}")
+	assert.Equal(t, Tail(2, []float64{}), []float64{}, "[]float64{}")
+	assert.Equal(t, Tail(2, []string{}), []string{}, "[]string{}")
+}
+
+func TestDrop(t *testing.T) {
```

Comment:
I wonder if there is a way to do different tests between `Drop` and `Skip`? Because `Skip` has that underlying array, wouldn't be possible to test that it's still present there by running some pointer checks -- and similarly, doing the same for `Drop` but making sure it's not there

Reply:
We currently don't have an `Unskip` function that would re-extend the underlying slice - but I don't know if its something idiomatic that we should expose. Also, I don't believe that we can really test that an element will get picked up by GC because by definition that means we can still access it so GC wouldn't sweep it. I think it is possible by some `unsafe` magic, but I don't know if its worth looking in to.

Reply:
Agree here, without `Unskip` and the unsafe nature of what we would want to do -- not worth it. You can skip it (haha)

Reply:
Resolved. Squashing.